src/Plugin/media/Source/MediaAvPortalVideoSource.php

```diff
@@ -15,4 +15,52 @@
  *   default_thumbnail_filename = "no-thumbnail.png",
  * )
  */
-class MediaAvPortalVideoSource extends MediaAvPortalSourceBase {}
+class MediaAvPortalVideoSource extends MediaAvPortalSourceBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getSupportedUrlFormats(): array {
+    return [
+      'https://ec.europa.eu/avservices/video/player.cfm?sitelang=en&ref=[REF]',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getSupportedUrlPatterns(): array {
+    return [
+      '@ec\.europa\.eu/avservices/video/player\.cfm\?(.+)@i',
+      '@ec\.europa\.eu/avservices/play\.cfm\?(.+)@i',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function transformUrlToReference(array $url): ?string {
+    preg_match('/(\d+)/', $url['query']['ref'], $matches);
+
+    // The reference should be in the format I-xxxx where x are numbers.
+    // Sometimes no dash is present, so we have to normalise the reference
+    // back.
+    if (isset($matches[0])) {
+      return 'I-' . $matches[0];
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function transformReferenceToUrl(string $reference): ?string {
+
+    $formats = $this->getSupportedUrlFormats();
```

Comment:
This should return a string in all cases, even if no matches are found.

Also, no need for empty line at the beginning of the method.